bundles/org.openhab.binding.shelly/src/main/java/org/openhab/binding/shelly/internal/handler/ShellyBaseHandler.java

```diff
@@ -530,6 +562,25 @@ private void fillDeviceStatus(ShellySettingsStatus status, boolean updated) {
         }
     }
 
+    /**
+     * Check if device has restarted and needs a new Thing initialization
+     *
+     * @return true: restart detected
+     */
+
+    private boolean checkRestarted(ShellySettingsStatus status) {
+        if (profile.isInitialized()
+                && ((status.uptime < stats.lastUptime) || (!profile.status.update.oldVersion.isEmpty()
+                        && !status.update.oldVersion.equals(profile.status.update.oldVersion)))) {
```

Comment:
Please try to overcome putting more brackets as necessary around expressions.

There's an eclipse save action to remove unncessary brackets:
![grafik](https://user-images.githubusercontent.com/1894841/109536841-7f1e5000-7abe-11eb-978e-59eb423dbfa8.png)

```suggestion
                && (status.uptime < stats.lastUptime || !profile.status.update.oldVersion.isEmpty()
                        && !status.update.oldVersion.equals(profile.status.update.oldVersion))) {
```

Reply:
I changed the setting, saved the file, but brackets still stay
?

Reply:
option activated, but adjustment was not automatically performed, so I changed this one manually
I made my experiences in the past with missing brackets..., so at some time I decided to take the safe path :-)

Reply:
You need to make some change in the file to trigger the save actions.

Reply:
I did, I'll have an eye on it